### PR TITLE
Add a Dockerfile that installs the project

### LINKED ISSUE #10: Add a `Dockerfile` that builds this project
We will be deploying this on kubernetes, so we need to build a `Dockerfile` that can be used to deploy this.

Use the official python docker image as the base image. `gunicorn` should be included in the image so it can be used to actually run the server. 

Per the [Linux FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard), let's put the code for this project under `/opt` - maybe under `/opt/unnamed-thingity-thing` (to be renamed later)

## Definition of done

- [x] Docker image builds with `docker build` when tried locally
- [x] Doing a `docker run --entrypoint /bin/bash -p 8000:8000 -it <image-name>` brings up a shell
- [x] `pip list` in the shell lists all the packages needed for the software to run
- [x] Running `python manage.py migrate` creates a base db inside the container
- [x] Running `gunicorn` with appropriate parameters servers the django app, and it can be accessed in the host machine via a browser on port 8000 (this was what the `-p` in the `docker run` command was for)

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Closes #10

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM python:3.12.5-slim-bullseye
+
+RUN apt-get update && apt-get install -y git
+
+RUN pip3 install gunicorn
+
+RUN git clone https://github.com/2i2c-org/unnamed-thingity-thing /opt
```
Comment (yuvipanda) [inline Dockerfile:7]: I think this should be `COPY . /opt`

Comment (GeorgianaElena) [reply Dockerfile:7]: Hmm, I'm not sure I understand @yuvipanda. Is there anything in the current directory that needs to be copied? I did a `ls` in the container and the repo got cloned in `/opt`. Maybe it should be `git clone https://github.com/2i2c-org/unnamed-thingity-thing /opt/unnamed-thingity-thing` instead?

Screenshot 2024-08-20 at 11 08 04

Comment (GeorgianaElena) [reply Dockerfile:7]: The answer was here all along :D 


Screenshot 2024-08-20 at 11 14 09

Comment (yuvipanda) [reply Dockerfile:7]: The `git clone` will clone the current main, so any builds during PRs won't work. Same for local testing.

**Pull request merged**